WordPress/openverse
# Fix i18n setup in the unit tests

## Fixes



Fixes #2050 by @obulat

## Description


This PR correctly sets up i18n in unit tests so that the keys are actually translated, both in the `setup` function that uses `useContext().i18n` from `Nuxt`, and in the templates' `$t` function.

Over the time, we have used several ways of mocking i18n in unit tests, and i18n is  set up slightly differently for different tests. This PR simplifies the set up by moving all of it into 3 points:
- `i18n` instance is set up in `test-utils/i18n.js`, and imported from there
- this instance is used in the `$nuxt` context mock - to be used everywhere where we use `useI18n()` in the components' `setup` functions.
- this instance is also used in the `render` function that mounts the component for tests. This is passed to `Vue` and is used in the templates' `$t` function.

All of the `i18n` set up and mocking from the components was removed.
This PR also uses `test-utils/render` for all components to correctly pass the `i18n` to all of the components. Using `render` from `testing-library` would require us to set up `i18n` separately (which is not necessary, and would needlessly complicate the set up).

## Testing Instructions

All the unit tests should pass. They should use the translated strings instead of the keys for selecting elements (see how the element selectors in the component tests have changed from e.g. "audio-details.related-audios" key to the value we can see in the UI ("Related audio")).


## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
